### DISCUSSION TITLE
`list[tuple]` is not implicitly type inferred as expected

### DISCUSSION THREAD

**Discussion by takuminakaso** [microsoft/pyright#7490] [Q&A] [answered]
As shown in the following code, `[("a", "b"), ("c", "d")]` is implicitly type inferred as `list[Unknown]`. Is this the intended behavior? I tested with pyright v1.1.352 and python v3.12.2

```python
from typing import Literal

# As expected, the following passes pyright.
x1: list[tuple[str, str]] = [("a", "b"), ("c", "d")]

# As expected, the following passes pyright.
x2: list[tuple[Literal["a"], Literal["b"]] | tuple[Literal["c"], Literal["d"]]] = [("a", "b"), ("c", "d")]

# As expected, the following doesn't pass pyright.
x3: list[tuple[int, int]] = [("a", "b"), ("c", "d")]

# Contrary to expectations, the following passes pyright.
# Pyright infers `x4` as `list[Unknown]`.
x4 = [("a", "b"), ("c", "d")]
x5: list[tuple[int, int]] = x4

```

**Comment by erictraut** [comment, marked as answer]
Yes, this is the intended behavior if `strictListInference` isn't enabled. For details about how pyright infers list expressions, refer to [this documentation](https://microsoft.github.io/pyright/#/type-inference?id=list-expressions). For tuple inference rules, refer to [this documentation](https://microsoft.github.io/pyright/#/type-inference?id=tuple-expressions).

**Reply by takuminakaso** [reply]
Thank you for presenting the documentations. Enabling `strictListInference` has resulted in the behavior I expect.
